tests/admin/test-class-my-yoast-route.php

```diff
@@ -0,0 +1,299 @@
+			->getMock();
+
+		$instance
+			->expects( $this->once() )
+			->method( 'is_myyoast_route' )
+			->will( $this->returnValue( true ) );
+
+		$instance
+			->expects( $this->once() )
+			->method( 'can_access_route' )
+			->will( $this->returnValue( true ) );
+
+		$instance
+			->expects( $this->once() )
+			->method( 'is_valid_action' )
+			->will( $this->returnValue( false ) );
+
+		$instance->register_hooks();
+
+		$this->assertFalse( has_action( 'admin_menu', array( $instance, 'register_route' ) ) );
+		$this->assertFalse( has_action( 'admin_init', array( $instance, 'handle_route' ) ) );
+	}
+
+	/**
+	 * Tests registrations of the hooks with every condition being okay.
+	 *
+	 * @covers WPSEO_MyYoast_Route::register_hooks
+	 */
+	public function test_register_hooks() {
+		$instance = $this
+			->getMockBuilder( 'WPSEO_MyYoast_Route' )
+			->setMethods( array( 'is_myyoast_route', 'can_access_route', 'is_valid_action' ) )
+			->getMock();
+
+		$instance
+			->expects( $this->once() )
+			->method( 'is_myyoast_route' )
+			->will( $this->returnValue( true ) );
+
+		$instance
+			->expects( $this->once() )
+			->method( 'can_access_route' )
+			->will( $this->returnValue( true ) );
+
+		$instance
+			->expects( $this->once() )
+			->method( 'is_valid_action' )
+			->will( $this->returnValue( true ) );
+
+		$instance->register_hooks();
+
+		$this->assertNotFalse( has_action( 'admin_menu', array( $instance, 'register_route' ) ) );
+		$this->assertNotFalse( has_action( 'admin_init', array( $instance, 'handle_route' ) ) );
+	}
+
+	/**
+	 * Tests handling the route.
+	 *
+	 * @covers WPSEO_MyYoast_Route::handle_route
+	 */
+	public function test_handle_route() {
+		$instance = $this
+			->getMockBuilder( 'WPSEO_MyYoast_Route' )
+			->setMethods( array( 'get_action', 'connect' , 'stop_execution' ) )
+			->getMock();
+
+		$instance
+			->expects( $this->once() )
+			->method( 'get_action' )
+			->will( $this->returnValue( 'connect' ) );
+
+		$instance
+			->expects( $this->once() )
+			->method( 'connect' );
+
+		$instance->handle_route();
+	}
+
+	/**
+	 * Tests the is myyoast route check.
+	 *
+	 * @covers WPSEO_MyYoast_Route::is_myyoast_route
+	 */
+	public function test_is_myyoast_route() {
+		$instance = new WPSEO_MyYoast_Route_Double();
+
+		$this->assertTrue( $instance->is_myyoast_route( 'wpseo_myyoast' ) );
+		$this->assertFalse( $instance->is_myyoast_route( 'another_route' ) );
+	}
+
+	/**
+	 * Tests if the given action is a valid action.
+	 *
+	 * @covers WPSEO_MyYoast_Route::is_valid_action
+	 */
+	public function test_is_valid_action() {
+		$instance = new WPSEO_MyYoast_Route_Double();
+
+		$this->assertTrue( $instance->is_valid_action( 'connect' ) );
+		$this->assertFalse( $instance->is_valid_action( 'invalid_action' ) );
+	}
+
+	/**
+	 * Tests connecting with having a saved clientid.
+	 *
+	 * @covers WPSEO_MyYoast_Route::connect
+	 */
+	public function test_connect_with_a_saved_clientid() {
```

Comment:
Can be removed as regardless of whether any info was saved a new client ID should be generated.